### ISSUE TITLE
Additional fields for film (and tv series)

### ISSUE THREAD

**Issue by simifilm** [citation-style-language/zotero-bits#50]
Film support is currently lacking. Things we need:
- Country of origin, required by many styles.
- Additional title. Depending on the situation you either need to have the original title and the international title, or the international title and the translated title. MLA requires this for example.
- MLA also requires distributor.
- Some styles give the production company.

For tv series, if the whole series and not a single episode is cited, often the tv station the show first aired and the producers are given.

**Comment by dstillman** [comment]
Related to the title issue: https://github.com/zotero/translators/pull/2069#issuecomment-565958735

Do we think a Translated Title field makes sense?

**Comment by bwiernik** [comment]
@dstillman Yes, a translated title field makes sense. The use for "Translated Title" would be for translations of a title into the language of the bibliography. So, an item might be cited as:
`Le petit prince [The little prince]`

For items where a translated version was consulted/is being cited, you would cite the translation, so the translation's title would go into `title` and the original title in a different language would go into `original-title`. 

Jurism/CSLm has `alt-title` that holds translations and transliterations of titles into various locales. Zotero could likely suffice with just one `translated-title` field. 

CSL will likely adopt a similar variable for name parts (https://github.com/citation-style-language/csl-evolution/issues/29), and is likely to adopt an `alt-title` or `translated-title` variable.

**Comment by denismaier** [comment]
> Yes, a translated title field makes sense. The use for "Translated Title" would be for translations of a title into the language of the bibliography. So, an item might be cited as:
> `Le petit prince [The little prince]`

At best, this translated title could be set on a per document basis, or you will want to have a way to indicate the language of the translation. Depending on the language of your bibliography you'll use a different translation. That's probably important for international scholars who publish in English and in their native language.

**Comment by bwiernik** [comment]
@denismaier That's essentially how Jurism's multilingual fields work. In a document, the user specifies the locale of the bibliography, and the `alt-` forms of fields for that locale are rendered. 

My impression is that the most common case is authors publishing in two languages—English and their local language. For that case, a simple `translated-title` field combined with a CSL style that does/does not include `translated-title` would suffice. That would avoid the complexity of Jurism's implementation which, while very flexible, is fairly confusing to work with. For the more complex cases of scholars working in many languages or needing fully-multilingual bibliographies, more functionality could likely be added as a plugin to Zotero. Talking with @fbennett about this a while back, it seemed like modularizing the multilingual components of Jurism into a plugin would be possible.

**Comment by denismaier** [comment]
> My impression is that the most common case is authors publishing in two languages—English and their local language. For that case, a simple `translated-title` field combined with a CSL style that does/does not include `translated-title` would suffice.

Yes, that is probably the most common case. But the source can be neither in English nor in your local language. Like if I write as a German scholar about "Le petit prince" once in German and once in English. Will a simple `translated-title` be enough?

> That's essentially how Jurism's multilingual fields work. In a document, the user specifies the locale of the bibliography, and the `alt-` forms of fields for that locale are rendered.

Well, but multilingual fields and `alt-` forms are not really the same thing... (For one, you usually don't have access to the variants in styles.)

>For the more complex cases of scholars working in many languages or needing fully-multilingual bibliographies, more functionality could likely be added as a plugin to Zotero. Talking with @fbennett about this a while back, it seemed like modularizing the multilingual components of Jurism into a plugin would be possible.

That sounds very promising and super-useful. Could be a neat way to bring much of Jurism's additional features back to vanilla Zotero. Unfortunately, I can't really help here :-(

**Comment by bwiernik** [comment]
> Well, but multilingual fields and alt- forms are not really the same thing... (For one, you usually don't have access to the variants in styles.)

Multilingual fields are implemented and rendered with `alt-` forms:
https://citeproc-js.readthedocs.io/en/latest/csl-m/#cs-alternative-extension

And `alt-` forms of title, container-title, translator, and issued are directly accessible in CSLm styles.
https://citeproc-js.readthedocs.io/en/latest/csl-m/#alt-title-extension

**Comment by denismaier** [comment]
Not sure, but as the docs say: `alt-` forms are used for reprints, translations and so on.
Regarding multilingual fields: I was talking about the mechanism described [here](https://github.com/Juris-M/jurism-docs/blob/master/tutorial.rst#multilingual-citations-in-an-english-article).
See also this test: https://github.com/Juris-M/citeproc-js/blob/master/fixtures/local/multilingual_ShortTitleVariant.txt

**Comment by bwiernik** [comment]
From https://citeproc-js.readthedocs.io/en/latest/csl-m/#cs-alternative-extension

> Elements within the scope of cs:alternative are rendered only when the value of language contains two valid ISO language codes joined by < or > (to indicate the direction of translation). The two language codes may be identical.

The interface you are linking to is how `cs:alternative` links up with specific locale forms of the fields.

**Comment by denismaier** [comment]
Honestly, I have to admit I'm a bit lost here. I just went through a couple of tests to understand how things work, through citeproc-js's changelog and through citeproc-js's integrator manual:

1. The `cs:alternative` mechanism is a relatively recent addition ([added in 1.1.216](https://github.com/Juris-M/citeproc-js/blob/9111953a18ec32ca4f6b2d8a3738c2c8b0ca9e37/CHANGES-1.1.txt)). But the multilingual mechanism for rendering field variants is older.

2. The data format for multilingual variants is described [here](https://github.com/Juris-M/citeproc-js/blob/b6a8778f17eed3037941e8e645567e8ec26989cc/attic/citeproc-doc.rst#data-format).

3. Here are some tests that show ho multilingual variants work. [One](https://github.com/Juris-M/citeproc-js/blob/master/fixtures/local/multilingual_NameSort.txt), [another one](https://github.com/Juris-M/citeproc-js/blob/master/fixtures/local/multilingual_ShortTitleVariant.txt).
There's no `cs:alternative` in either case. 
In one case, the variants are controlled by the `default-locale="en-US-x-sort-ja-Hrkt">`
In the other, there's a special `LANGPARAMS` section in the test fixture.

4. This [test](https://github.com/Juris-M/citeproc-js/blob/dcebdd9bf612b6b418cf102601146f635f417a2a/fixtures/local/name_TranslatorInMultiLayout.txt) and [that one](https://github.com/Juris-M/citeproc-js/blob/a0d4d48701271c4660b45ac79794d40d5710fcda/fixtures/local/jasoc_example07.txt) show how the `cs:alternative` mechanism works. Interestingly, the translated title is saved as a multi-key, but not as `alt-title`. (So they are indeed more closely related than I thought.)
The other variables appear as `alt-` prefixed variables. 
Just [tested](https://gist.github.com/denismaier/1ea6fb94e0730829900df10a9f075d3c): If I add `alt-title`, it will use this instead of the variant from the `multi` variable. So, the variants in `multi` seem to be a fallback for `alt-title`? (But will this also work the other way round?)

Perhaps @fbennett can provide some clarifying hints...

**Comment by denismaier** [comment]
@bwiernik Can we add the Zotero label now that we've almost sorted out the `translated-title` issue.

**Comment by simifilm** [comment]
It's more than half a decade later, and it's still not possible to assign the country of origin to a film.

**Comment by bwiernik** [comment]
@simifilm You can currently enter the country in Zotero in Extra like this: `Publisher place: Toronto, Canada`
